### ISSUE TITLE
Shedlock is Not Considering The Consistency Specified In CassandraLockProvider

### ISSUE THREAD

**Issue by DebajitKumarPhukan** [lukas-krecan/ShedLock#357]
### Problem Statement:
By default  shedlock is taking "QUORUM" as consistency. Even after specifying consistency level as "ONE" in the CassandraLockProvider() overloaded constructor.

### StackTrace:
com.datastax.oss.driver.api.core.AllNodesFailedException: All 1 node(s) tried for the query failed (showing first 1 nodes, use getAllErrors() for more): Node(endPoint=192.###.99.###/192.###.99.##:####, hostId=eb2a990d-2f95-40fd-bf72-48f2422ce047, hashCode=12da0bf): [com.datastax.oss.driver.api.core.servererrors.UnavailableException: Not enough replicas available for query at consistency QUORUM (2 required but only 1 alive)]
	at com.datastax.oss.driver.api.core.AllNodesFailedException.copy(AllNodesFailedException.java:141) ~[java-driver-core-4.9.0.jar:na]
	at com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures.getUninterruptibly(CompletableFutures.java:149) ~[java-driver-core-4.9.0.jar:na]
	at com.datastax.oss.driver.internal.core.cql.CqlRequestSyncProcessor.process(CqlRequestSyncProcessor.java:53) ~[java-driver-core-4.9.0.jar:na]
	at com.datastax.oss.driver.internal.core.cql.CqlRequestSyncProcessor.process(CqlRequestSyncProcessor.java:30) ~[java-driver-core-4.9.0.jar:na]
	at com.datastax.oss.driver.internal.core.session.DefaultSession.execute(DefaultSession.java:230) ~[java-driver-core-4.9.0.jar:na]
	at com.datastax.oss.driver.api.core.cql.SyncCqlSession.execute(SyncCqlSession.java:54) ~[java-driver-core-4.9.0.jar:na]
	at net.javacrumbs.shedlock.provider.cassandra.CassandraStorageAccessor.execute(CassandraStorageAccessor.java:153) ~[shedlock-provider-cassandra-4.16.0.jar:na]
	at net.javacrumbs.shedlock.provider.cassandra.CassandraStorageAccessor.insert(CassandraStorageAccessor.java:111) ~[shedlock-provider-cassandra-4.16.0.jar:na]
	at net.javacrumbs.shedlock.provider.cassandra.CassandraStorageAccessor.insertRecord(CassandraStorageAccessor.java:51) ~[shedlock-provider-cassandra-4.16.0.jar:na]
	at net.javacrumbs.shedlock.support.StorageBasedLockProvider.doLock(StorageBasedLockProvider.java:81) ~[shedlock-core-4.16.0.jar:na]
	at net.javacrumbs.shedlock.support.StorageBasedLockProvider.lock(StorageBasedLockProvider.java:65) ~[shedlock-core-4.16.0.jar:na]
	at net.javacrumbs.shedlock.core.DefaultLockingTaskExecutor.executeWithLock(DefaultLockingTaskExecutor.java:63) ~[shedlock-core-4.16.0.jar:na]
	at net.javacrumbs.shedlock.core.DefaultLockingTaskExecutor.executeWithLock(DefaultLockingTaskExecutor.java:54) ~[shedlock-core-4.16.0.jar:na]
	at net.javacrumbs.shedlock.core.DefaultLockingTaskExecutor.executeWithLock(DefaultLockingTaskExecutor.java:43) ~[shedlock-core-4.16.0.jar:na]
	at com.ricky.practice.scheduler.LoginScheduler.logLoginDetails(LoginScheduler.java:58) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_05]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_05]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_05]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[na:1.8.0_05]
	at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84) ~[spring-context-5.3.1.jar:5.3.1]
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-5.3.1.jar:5.3.1]
	at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:95) [spring-context-5.3.1.jar:5.3.1]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_05]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_05]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_05]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.8.0_05]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_05]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_05]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_05]
	Suppressed: com.datastax.oss.driver.api.core.servererrors.UnavailableException: Not enough replicas available for query at consistency QUORUM (2 required but only 1 alive)

### Dependencies:
<dependency>
<groupId>net.javacrumbs.shedlock</groupId>
<artifactId>shedlock-spring</artifactId>
<version>4.16.0</version>
</dependency>

<dependency>
<groupId>net.javacrumbs.shedlock</groupId>
<artifactId>shedlock-provider-cassandra</artifactId>
<version>4.16.0</version>
</dependency>

**Comment by lukas-krecan** [comment]
Hi, thanks for feedback. I am not Cassandra expert, the provider was contributed by someone else. I have checked the code, and the consistency level is applied [here](https://github.com/lukas-krecan/ShedLock/blob/master/providers/cassandra/shedlock-provider-cassandra/src/main/java/net/javacrumbs/shedlock/provider/cassandra/CassandraStorageAccessor.java#L153). Do you have any ideas what could be wrong there?

**Comment by DebajitKumarPhukan** [comment]
@lukas-krecan Thanks for the response. Yes, I already saw that and even I went into puzzles after seeing the "Consistency" level not getting honored. I went through the entire request life cycle and everything looked good. After tracing the request from CqlRequestSyncProcessor to Cassandra db, I identified that the Cassandra db also has its own priority based on which it decides if it is going to honor the "Consistency" level specified in the driver request. If the db has default "Consistency" set to Quorum, it might discard the java driver's "Consistency" level.  May be a bug in Cassandra.
